src/main/java/org/opentripplanner/routing/impl/ATLFareServiceImpl.java

```diff
@@ -0,0 +1,373 @@
+                return RideType.XPRESS;
+            case MARTA_AGENCY_ID:
+                return RideType.MARTA;
+            case GCT_AGENCY_ID:
+                if (shortName.equalsIgnoreCase("102") ||
+                    shortName.equalsIgnoreCase("103A") ||
+                    shortName.equalsIgnoreCase("110")) {
+                    return RideType.GCT_EXPRESS_Z1;
+                } else if (shortName.equalsIgnoreCase("101") ||
+                    shortName.equalsIgnoreCase("103")) {
+                    return RideType.GCT_EXPRESS_Z2;
+                }
+                return RideType.GCT_LOCAL;
+            case STREETCAR_AGENCY_ID:
+                return RideType.STREETCAR;
+            default:
+                return RideType.MARTA;
+        }
+    }
+
+    private static TransferMeta classifyTransfer(RideType toRideType, RideType fromRideType, Fare.FareType fareType) {
+        switch (toRideType) {
+            case STREETCAR:
+                return new TransferMeta(TransferType.NO_TRANSFER);
+            case COBB_LOCAL:
+                if (!isElectronicPayment(fareType)) {
+                    if (fromRideType == RideType.COBB_LOCAL || fromRideType == RideType.COBB_EXPRESS) {
+                        return new TransferMeta(TransferType.FREE_TRANSFER, 180, 4);
+                    }
+                    return new TransferMeta(TransferType.END_TRANSFER);
+                }
+                switch(fromRideType) {
+                    case COBB_LOCAL:
+                    case COBB_EXPRESS:
+                    case MARTA:
+                        return new TransferMeta(TransferType.FREE_TRANSFER, 180, 4);
+                    default:
+                        return new TransferMeta(TransferType.END_TRANSFER);
+                }
+            case COBB_EXPRESS:
+                if (!isElectronicPayment(fareType)) {
+                    switch(fromRideType) {
+                        case COBB_EXPRESS:
+                            return new TransferMeta(TransferType.FREE_TRANSFER, 180, 4);
+                        case COBB_LOCAL:
+                            return new TransferMeta(TransferType.TRANSFER_PAY_DIFFERENCE, 180, 4);
+                        default:
+                            return new TransferMeta(TransferType.END_TRANSFER);
+                    }
+                }
+                // Electronic payment
+                switch(fromRideType) {
+                    case COBB_EXPRESS:
+                    case MARTA:
+                        return new TransferMeta(TransferType.FREE_TRANSFER, 180, 4);
+                    case COBB_LOCAL:
+                        return new TransferMeta(TransferType.TRANSFER_PAY_DIFFERENCE, 180, 4);
+                    default:
+                        return new TransferMeta(TransferType.NO_TRANSFER);
+                }
+            case MARTA:
+               if (!isElectronicPayment(fareType)) return new TransferMeta(TransferType.END_TRANSFER);
```

Comment:
There is one exception here, I am not sure how we want to implement it. If a trip involves MARTA Rail, passengers must pay with Breeze, so transfers that follow the leg on MARTA Rail are free.

Reply:
(Per discussion, will not pursue)